src/domain/chainContext.ts

```diff
@@ -294,6 +323,26 @@ export class ChainContext {
       }
     }
   }
+
+  get status(): ChainStatus {
```

Comment:
OK, I've gone through and actually made it so that it will self-detect if it's running within a kubernetes environment. If it detects this, the watch-dog won't exit, but will instead set the chain sync to `UNKNOWN` status, leading the `/health` API endpoint to error with `503`, which can be picked up by directing the liveliness probe to `/health`.